### ISSUE TITLE
[Help]: Invalid curl command "json.decoder.JSONDecodeError:" in /completions and /chat/completions

### ISSUE THREAD

**Issue by VfBfoerst** [BerriAI/litellm#1703]
### What happened?

A bug happened!   
  
Cool project!   
  
My bug is reproducable with:  
- newly installed module `pip install 'litellm[proxy]'`
- Docker-Image ghcr.io/berriai/litellm:main-latest  
  
I want to proxy to remote Ollama instance. I tried different approaches while starting the proxy, which was e.g.:   
  
`litellm --model ollama/mistral --api_base http://123.123.123.123:10000/ --host 0.0.0.0 --port 8000 --debug`
`litellm --model ollama/mistral-7b --api_base http://123.123.123.123:10000/ --host 0.0.0.0 --port 8000 --debug`
`litellm --model ollama/mistral:7b --api_base http://123.123.123.123:10000/ --host 0.0.0.0 --port 8000 --debug`  
`litellm --model ollama/mistral:7b --api_base http://123.123.123.123:10000/generate --host 0.0.0.0 --port 8000 --debug`    
  
If I curl or use the completions endpoints, I get always the jsondecode error.  
  
Thank you :-)

### Relevant log output

```shell
INFO:     10.0.2.100:39144 - "POST /completions?model=mistral-7b HTTP/1.1" 500 Internal Server Error

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/litellm/proxy/proxy_server.py", line 1794, in completion
    data = json.loads(body_str)
  File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
```

_Edit: I also checked the logs of my ollama instance. There is no single request logged, so the issue must be anytime before._

**Comment by ghost** [comment]
>  File "/usr/local/lib/python3.9/site-packages/litellm/proxy/proxy_server.py", line 1794, in completion
    data = json.loads(body_str)

@VfBfoerst looks like the curl command being made is not a valid call 

just run `litellm --test` and see if it works 

https://docs.litellm.ai/docs/proxy/quick_start#quick-start---litellm-proxy-cli

**Comment by VfBfoerst** [comment]
Thanks, could be the same issue like #1458. I will close this issue for now and take a look into the other issue, maybe my problem will be solved there too. Thank you!

**Comment by VfBfoerst** [comment]
#1458 and PR [1497](https://github.com/BerriAI/litellm/pull/1497) did not fix my issue. Same behavior: There is no request put to my remote ollama instance.   
Current config.yaml:  
```yaml
model_list:
  - model_name: llama2
    litellm_params:
      model: ollama/llama2
      api_base: http://123.123.123.123:8080
      stream: True
```
No matter which model I tried, I am always getting the JSON-Decode error. I already enabled `--detailed_debug` but got no conclusion.  
After starting the litellm proxy, trying with `litellm --test` results in:  
  
- missing API Key (litellm --test --host 123.123.123.123): 
```python
openai.InternalServerError: Error code: 500 - {'error': {'message': 'The api_key client option must be set either by passing api_key to the client or by setting the OPENAI_API_KEY environment variable\n\nTraceback (most recent call last):\n  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 6289, in exception_type\n    message = original_exception.message\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^\nAttributeError: \'OpenAIError\' object has no attribute \'message\'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 2350, in chat_completion\n    response = await litellm.acompletion(**data)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 2897, in wrapper_async\n    raise e\n  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 2740, in wrapper_async\n    result = await original_function(*args, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/litellm/main.py", line 287, in acompletion\n    raise exception_type(\n          ^^^^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 7504, in exception_type\n    raise original_exception\n  File "/usr/local/lib/python3.11/site-packages/litellm/main.py", line 274, in acompletion\n    response = await init_response\n               ^^^^^^^^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 424, in acompletion\n    raise e\n  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 387, in acompletion\n    openai_aclient = AsyncOpenAI(\n                     ^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/openai/_client.py", line 296, in __init__\n    raise OpenAIError(\nopenai.OpenAIError: The api_key client option must be set either by passing api_key to the client or by setting the OPENAI_API_KEY environment variable\n', 'type': 'None', 'param': 'None', 'code': 500}}
``` 
Same with `litellm --test --host 123.123.123.123 --api_base 123.123.123.123:8080`  
  
Both, litellm and Ollama, are running inside docker containers.  
  
Is there anything else i could test?

**Comment by amirkhonov** [comment]
I have the same issue:
```
curl -X 'POST' \
  'http://0.0.0.0:8000/engines/ollama%2Fdeepseek-coder%3A6.7b/completions' \
  -H 'accept: application/json' \
  -d ''
 ```
The response:
```
{
  "error": {
    "message": "Expecting value: line 1 column 1 (char 0)\n\nTraceback (most recent call last):\n  File \"/Users/desktop/.pyenv/versions/3.11.7/lib/python3.11/site-packages/litellm/proxy/proxy_server.py\", line 2125, in completion\n    data = json.loads(body_str)\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/desktop/.pyenv/versions/3.11.7/lib/python3.11/json/__init__.py\", line 346, in loads\n    return _default_decoder.decode(s)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/desktop/.pyenv/versions/3.11.7/lib/python3.11/json/decoder.py\", line 337, in decode\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/desktop/.pyenv/versions/3.11.7/lib/python3.11/json/decoder.py\", line 355, in raw_decode\n    raise JSONDecodeError(\"Expecting value\", s, err.value) from None\njson.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)\n",
    "type": "None",
    "param": "None",
    "code": 500
  }
}
```

**Comment by VfBfoerst** [comment]
I found the issue at least in my case. The "problem" was trying the requests only with the swagger ui. There, you can only set the model as parameter, e.g.:  
 
![image](https://github.com/BerriAI/litellm/assets/64251549/810e1415-4d19-49b9-b34c-0c675dc12f1c)
  
that led to the described jsondecode error. After inspecting the curl request, i saw the empty `-d` flag. Therefore with the Swagger UI, there will no data be sent to the api, which is the same case in @amirkhonov try.  
After simply trying a curl request with data, the proxy worked as expected:  
```bash
 curl http://localhost:8000/v1/chat/completions     -H "Content-Type: application/json"     -d '{
        "model": "llama2",
        "messages": [
            {
                "role": "system",
                "content": "You are a helpful assistant."
            },
            {
                "role": "user",
                "content": "tell me a short joke"
            }
        ]
    }'
```
Led to the answer:  
```bash
{"id":"chatcmpl-831a02b0-3040-495c-a3ba-8c56f5bf62f7","choices":[{"finish_reason":"stop","index":0,"message":{"content":"Of course! Here is a short and simple joke for you:\n\nWhy don't scientists trust atoms? Because they make up everything! 😂","role":"assistant"}}],"created":1708345168,"model":"ollama/llama2","object":"chat.completion","system_fingerprint":null,"usage":{"prompt_tokens":33,"completion_tokens":37,"total_tokens":70}}
```
  
_(Gracefully, we mustn't discuss the quality of the returned joke :})_  
  
**tl;dr**:
There must be data given to the curl request, which isn't possible in the swagger ui. The proxy seems to work correctly with ollama when data was given to the `-d` flag.  
  
Let me make the following suggestion:  
Add another field in the completions endpoint in the swagger UI, so `-d` can be filled correctly.

**Comment by VfBfoerst** [comment]
You can set it via swagger ui:  
![image](https://github.com/BerriAI/litellm/assets/64251549/7d89dddf-5d7f-4310-8da6-7d99c88cc5a4)
